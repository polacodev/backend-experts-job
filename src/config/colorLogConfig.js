import colors from 'colors';

const colorLog = colors.setTheme({
  success: 'green',
  warn: 'yellow',
  info: 'cyan',
  error: 'red',
});

export default colorLog;
