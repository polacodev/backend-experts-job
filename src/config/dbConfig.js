import mongoose from 'mongoose';

const MONGO_URL = 'mongodb://localhost:27017/ExpertsJob';
const database = async () => {
  try {
    await mongoose.connect(MONGO_URL,
      {
        useNewUrlParser: true,
        useUnifiedTopology: true,
        useFindAndModify: false,
      });
    console.log('DB Connection SUCCESSFULLY');
  } catch (error) {
    console.log('DB Connection FAILED');
  }
};

export default database;
