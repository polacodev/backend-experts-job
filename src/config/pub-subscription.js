import { connect } from 'mqtt';
import { MQTTPubSub } from 'graphql-mqtt-subscriptions';

const client = connect('mqtt://test.mosquitto.org', {
  reconnectPeriod: 1000,
});

const pubsub = new MQTTPubSub({
  client,
});
export default pubsub;
// import { PubSub } from 'apollo-server';

// const pubsub = new PubSub();

// export default pubsub;
