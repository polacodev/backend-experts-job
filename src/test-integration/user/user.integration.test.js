import querystring from 'querystring';

import { API_URL, api } from '../services.config';

describe('aaaa', () => {
  it('aaaa', async () => {
    const query = `
    query getAll {
       getUsers {
         ...User
       }
     }
    
     fragment User on UserType {
       _id
       name
       email
       workarea
       status
       cellphone
     }
  `;
    const response = await api.post(API_URL, querystring.stringify({ query }));
    expect(response).toMatchSnapshot();
  });
});
