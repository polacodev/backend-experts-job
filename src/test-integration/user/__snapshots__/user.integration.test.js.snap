// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aaaa aaaa 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "baseURL": "http://localhost:4000/graphql",
    "data": "query=%0A%20%20%20%20query%20getAll%20%7B%0A%20%20%20%20%20%20%20getUsers%20%7B%0A%20%20%20%20%20%20%20%20%20...User%0A%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%7D%0A%20%20%20%20%0A%20%20%20%20%20fragment%20User%20on%20UserType%20%7B%0A%20%20%20%20%20%20%20_id%0A%20%20%20%20%20%20%20name%0A%20%20%20%20%20%20%20email%0A%20%20%20%20%20%20%20workarea%0A%20%20%20%20%20%20%20status%0A%20%20%20%20%20%20%20cellphone%0A%20%20%20%20%20%7D%0A%20%20",
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
      "Content-Type": "application/x-www-form-urlencoded",
      "X-Custom-Header": "foobar",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "post",
    "timeout": 1000,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "url": "http://localhost:4000/graphql",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": Object {
    "data": Object {
      "getUsers": Array [],
    },
    "extensions": Object {
      "operation": null,
      "runTime": 8705,
    },
  },
  "headers": Object {
    "content-type": "application/json; charset=utf-8",
  },
  "request": XMLHttpRequest {},
  "status": 200,
  "statusText": "OK",
}
`;
